src/Common/src/Interop/Linux/System.Net.Security.Native/Interop.HeimdalNtlmException.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class NetSecurity
+    {
+        internal sealed class HeimdalNtlmException : Exception
+        {
+            public HeimdalNtlmException(string message) : base(message)
+            {
+            }
+
+            public HeimdalNtlmException(int error)
+                : base(SR.Format(SR.net_generic_heimntlm_operation_failed, error))
+            {
+                HResult = error;
+            }
+
+            public static void AssertOrThrowIfError(string message, int error)
```

Comment:
I suggest we just make this Throw.  Often Debug.Assert/Fail is used in situations where we're not throwing, but since we are here, you could just throw, and if you want extra debugging information, use Debug.WriteLine.  Additionally, it's strange that the message is being used only for the assert but not for the exception; that could be confusing for a developer calling this method, especially since it looks like there's useful information being captured in those messages and not conveyed via the exception.  So, I suggest:
1. Replace the Debug.Fail with Debug.WriteLine.
2. Either incorporate the message into the exception or remove the message parameter.
3. Change the name of the method to ThrowIfError.